### PR TITLE
imp(evm): update move gas functions to a new file

### PR DESCRIPTION
**Pull request by fedekunze**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/keeper/state_transition.go [hunk 1]
```diff
@@ -276,8 +273,10 @@ func (k *Keeper) ApplyTransaction(ctx sdk.Context, tx *ethtypes.Transaction) (*t
 		} else if commit != nil {
 			// PostTxProcessing is successful, commit the tmpCtx
 			commit()
-			// Since the post processing can alter the log, we need to update the result
+			// Since the post processing can alter the log and the gas used, we need
+			// to update the result
 			res.Logs = types.NewLogsFromEth(receipt.Logs)
+			res.GasUsed = receipt.GasUsed
```
Comment (fedekunze) [inline x/evm/keeper/state_transition.go:279]: @facs95 @yihuang if the gas used from the receipt is updated in the hook, the gas used was still the old one defined by the initial result

Comment (yihuang) [reply x/evm/keeper/state_transition.go:279]: I see, so now hooks can increase the gas used, cool.

Comment (fedekunze) [reply x/evm/keeper/state_transition.go:279]: the main issue is that the gas used in the hooks is not considered in the transaction. Gas estimation would also need to be updated to reflect this change.

Not sure if it's worth it...

Comment (yihuang) [reply x/evm/keeper/state_transition.go:279]: yeah, precompiles has an `EstimatedGas` method though, which is better in this regard.

File: x/evm/keeper/state_transition.go [hunk 2]
```diff
@@ -430,17 +432,21 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context, msg core.Message, trace
...
 	minGasMultiplier := k.GetMinGasMultiplier(ctx)
 	minimumGasUsed := gasLimit.Mul(minGasMultiplier)
 
 	if msg.Gas() < leftoverGas {
 		return nil, sdkerrors.Wrapf(types.ErrGasOverflow, "message gas limit < leftover gas (%d < %d)", msg.Gas(), leftoverGas)
 	}
+
 	temporaryGasUsed := msg.Gas() - leftoverGas
+
+	// gasUsed = max(gasUsed, gasLimit * minGasMultiplier)
```
Comment (facs95) [inline x/evm/keeper/state_transition.go:447]: Remove comment

**Review by facs95 — CHANGES_REQUESTED** [review]
Is this something that we still think is needed? Based on the conversation https://github.com/evmos/ethermint/pull/1299/files#r953968127

**Review by danburck — APPROVED** [review]
@fedekunze moving the gas functions LGTM 

It sounds like there are more changes required to account for gas used during post processing. Can you add a ticket for this please?

**Comment by danburck** [comment]
@fedekunze the integration tests are breaking. Can you add a description to the PR? Are there any expected changes in the amount of gasUsed?

File: x/evm/keeper/state_transition.go (hunk 1 above)
Comment (danburck) [reply x/evm/keeper/state_transition.go:280]: @fedekunze increasing the gasUsed by how much gas was used during Post-Processing is out of scope of this PR and should be treated as an extra ticket.

Comment (fedekunze) [reply x/evm/keeper/state_transition.go:280]: yup, agree

**Pull request closed without merging**